google/cloud/aiplatform/metadata/execution.py

```diff
@@ -55,28 +103,28 @@ def _create_resource(
         )
 
     @classmethod
-    def _update_resource(
-        cls, client: utils.MetadataClientWithOverride, resource: proto.Message,
-    ) -> proto.Message:
-        return client.update_execution(execution=resource)
-
-    def add_artifact(
-        self, artifact_resource_name: str, input: bool,
+    def _list_resources(
+        cls,
+        client: utils.MetadataClientWithOverride,
+        parent: str,
+        filter: Optional[str] = None,
     ):
-        """Creates a new Metadata resource.
+        """List Executions in the parent path that matches the filter.
 
         Args:
-            artifact_resource_name (str):
-                Required. The full resource name of the Artifact to connect to the Execution through an Event.
-            input (bool)
-                Required. Whether Artifact is an input event to the Execution or not.
+            client (utils.MetadataClientWithOverride):
+                Required. client to send require to Metadata Service.
+            parent (str):
+                Required. The path where Executions are stored.
+            filter (str):
+                Optional. filter string to restrict the list result
         """
 
-        event = Event(
-            artifact=artifact_resource_name,
-            type_=Event.Type.INPUT if input else Event.Type.OUTPUT,
-        )
+        list_request = ListExecutionsRequest(parent=parent, filter=filter,)
+        return client.list_executions(request=list_request)
 
-        self.api_client.add_execution_events(
-            execution=self.resource_name, events=[event],
-        )
+    @classmethod
+    def _update_resource(
+        cls, client: utils.MetadataClientWithOverride, resource: proto.Message,
+    ) -> proto.Message:
```

Comment:
Docstring

Reply:
done. also added docstring for this method for other resources.